week2/index.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+const fs = require('fs');
+
+const TODOS_FILENAME = process.env["TODOS_FILENAME"] || "to-do-list.json";
+
+let myTodoList = []
+
+
+// Helper functions definitions
+
+let readFileFunction = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("I'm sorry, friend, but that file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    console.log("That file exists but you don't have permission.");
+                } else {
+                    console.log(`Uh oh, an unknown error occured: ${err.message}`);
+                }
+                process.exit()
+            }
+            //  console.log(`Successfully read the file: ${filename}`)
+            resolve(data)
```

Comment:
Your promise doesn't reject ever.

(I know this is pretty much unchanged from my example - but I wasn't doing the homework for you, just giving you a starting point)

Reply:
i didnt know that every promise you need to reject it or catch, thanks for the feedback, i updated my code, i hope it is okay like this